src/ILCompiler/src/Program.cs

```diff
@@ -284,14 +288,24 @@ private int Run(string[] args)
                         entrypointModule = module;
                     }
 
-                    compilationRoots.Add(new ExportedMethodsRootProvider(module));
+                    if (!_isWasmCodegen)
```

Comment:
There should be a TODO saying we're only doing this because wasm can't do anything.

Reply:
Done